_pytest/assertion/__init__.py

```diff
@@ -151,9 +172,10 @@ def pytest_runtest_teardown(item):
 
 
 def pytest_sessionfinish(session):
-    hook = session.config._assertstate.hook
-    if hook is not None:
-        hook.session = None
+    assertstate = getattr(session.config, '_assertstate', None)
+    if assertstate:
+        if assertstate.hook is not None:
+            assertstate.hook.session = None
```

Comment:
Fair enough, we wondered this too but decided to leave the existing code since we where not changing functionality.  But will update.